### PR TITLE
Python 3.11 support with dill monkey patching

### PR DESCRIPTION
**Pull request by AnandInguva**
**Please** add a meaningful description for your change here

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by AnandInguva** [comment]
Run Python_Integration PreCommit

**Comment by tvalentyn** [comment]
Verified that this PR is identical to https://github.com/apache/beam/pull/26120 modulo comment-only changes to base_image_requirements_manual.txt.

**Comment by tvalentyn** [comment]
Merging based on test signal from https://github.com/apache/beam/pull/26120

**Review by tvalentyn — APPROVED** [review]

File: .test-infra/jenkins/job_PreCommit_Portable_Python.groovy [hunk 1]
```diff
@@ -40,15 +42,17 @@ builder.build {
...
     gradle {
       rootBuildScriptDir(commonJobProperties.checkoutDir)
-      tasks(':sdks:python:test-suites:portable:py37:preCommitPy37')
+      tasks(':sdks:python:test-suites:portable:py${lowestSupported}:preCommitPy${lowestSupported}')
       commonJobProperties.setGradleSwitches(delegate)
     }
     gradle {
       rootBuildScriptDir(commonJobProperties.checkoutDir)
-      tasks(':sdks:python:test-suites:portable:py310:preCommitPy310')
+      tasks(':sdks:python:test-suites:portable:${highestSupported}:preCommitPy${highestSupported}')
```
Comment (ahmedabu98) [inline .test-infra/jenkins/job_PreCommit_Portable_Python.groovy:55]: Looks like these changes are causing [Python Portable Precommit to fail](https://ci-beam.apache.org/job/beam_PreCommit_Portable_Python_Commit/27558/console) with:
```
Project 'py${lowestSupported}' not found in project ':sdks:python:test-suites:portable'
```
Need to use double quotes when using embedded values

Comment (ahmedabu98) [reply .test-infra/jenkins/job_PreCommit_Portable_Python.groovy:55]: Fix in #26151

**Pull request merged**